app/src/main/java/org/mozilla/reference/browser/browser/FirefoxAccountsIntegration.kt

```diff
@@ -63,29 +62,61 @@ class FirefoxAccountsIntegration(
     override val coroutineContext: CoroutineContext
         get() = Dispatchers.Main + job + exceptionHandler
 
-    lateinit var account: Deferred<FirefoxAccount>
+    lateinit var account: FirefoxAccount
 
-    var profile: Profile? = null
+    @Volatile var profile: Profile? = null
         private set
 
     @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
     fun init() {
         job = Job()
+        account = initAccount()
+    }
 
-        account = async {
-            getAuthenticatedAccount()?.let {
-                profile = it.getProfile(true).await()
-                return@async it
+    private fun initAccount(): FirefoxAccount {
```

Comment:
I think we should separate dealing with account from dealing with profile, otherwise we're mixing two different concepts together.

We can have an good account state without ever fetching a profile, and everything should generally function (e.g. sync will work, UI should degrade to something sensible - if it doesn't, that's a bug, etc).

In other words, profile presence isn't tied to a well-functioning account. If the profile server has an outage (or if we hit a network intermittent while talking to it), that shouldn't affect things negatively.

This mixing also makes code pretty confusing to read and inefficient. In authenticate, we read an account state (json dump) from disc, and initialize a FirefoxAccount object (which goes through JSON state parsing, state upgrade, etc path on the Rust side). Then, if we don't have a profile for some reason by the time we hit `authenticate`, we do all of this _again_!

In other words, we should only ever hit the expensive `authenticate` path if we don't have a valid account state. Currently, `SettingsFragment` checks for `profile` presence, which is wrong.